keras_nlp/samplers/beam_sampler_test.py

```diff
@@ -30,10 +30,10 @@ def setUp(self):
         self.length = 12
         self.vocab_size = len(self.int_lookup)
 
-        def next(prompt, state, index):
-            # Return a distribution favoring the next char in state.
-            logits = tf.one_hot(state[:, index], self.vocab_size) * 1e9
-            return logits, state
+        def next(prompt, cache, index):
+            # Return a distribution favoring the next char in cache.
+            logits = tf.one_hot(cache[:, index], self.vocab_size) * 1e9
```

Comment:
I might actually just get in the habit of making a `hidden_state` as we will need to do this for guides and examples.
```
# Return a distribution favoring the next char in cache.
hidden_state = cache[:, index]
logits = tf.one_hot(hidden_state self.vocab_size) * 1e9
```

Reply:
returning a dummy `hidden_states` sounds good to me, i will actually create a tensor as users may have the impression `hidden_states` are coupled with `cache` by the code above.